### PR TITLE
Brevity#274 input of

### PR DESCRIPTION
**Pull request by ixmanuel**
Trial for #274 - InputOf

### CODE REVIEW

File: src/main/java/org/cactoos/io/InputOf.java [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+     */
+    private final Input origin;
+
+    /**
+     * Ctor.
+     *
+     * @param src The file
+     */
+    public InputOf(final File src) {
+        this(new FileAsInput(src));
```
Comment (fabriciofx) [inline src/main/java/org/cactoos/io/InputOf.java]: @ixmanuel You can't use lambda function here? I thought in something like:
```
public InputOf(final File src) {
  this(src -> new FileInputStream(src));
}
```

File: src/main/java/org/cactoos/text/RegexText.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
+/**
```
Comment (fabriciofx) [inline src/main/java/org/cactoos/text/RegexText.java]: @ixmanuel WDYT to create a new PR only with Regex classes code?

File: src/test/java/org/cactoos/text/RegexTextTest.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
+/**
```
Comment (fabriciofx) [inline src/test/java/org/cactoos/text/RegexTextTest.java]: @ixmanuel The same above.

File: src/main/java/org/cactoos/text/RegexText.java (hunk 2 above)
Comment (ixmanuel) [inline src/main/java/org/cactoos/text/RegexText.java]: Thanks. I've already fixed it.

File: src/test/java/org/cactoos/text/RegexTextTest.java (hunk 3 above)
Comment (ixmanuel) [inline src/test/java/org/cactoos/text/RegexTextTest.java]: Thanks. I've already fixed it.

File: src/main/java/org/cactoos/io/InputOf.java (hunk 1 above)
Comment (ixmanuel) [inline src/main/java/org/cactoos/io/InputOf.java]: Yes, I tried according to the example that you exposed in the conversation # 312 but I didn't see more benefits for the rest of the cases, so I went back to the idea of Decorating Envelopes and thus, I didn't change anything, I have only created InputOf.

**Comment by fabriciofx** [comment]
@ixmanuel Could you resubmit `RegexText` as a new issue/PR? Thanks in advance!

**Comment by yegor256** [comment]
@ixmanuel I don't like that `new InputOf(String)` would work as a `ResourceAsInput`. That's counter intuitive. I would expect it to do what `BytesAsInput` is doing now: convert a string content to the `Input`.

**Comment by yegor256** [comment]
@ixmanuel this is what I would expect:

  * `InputOf(File)`
  * `InputOf(Path)`
  * `InputOf(URL)`
  * `InputOf(URI)`
  * `InputOf(String)`
  * `InputOf(Text)`
  * `InputOf(Bytes)`
  * `InputOf(byte[])`
  * `InputOf(char[])`
  * `InputOf(StringBuilder)`
  * `InputOf(StringBuffer)`
  
maybe something else. "Resource" is not an option. We still need to have `ResourceAsInput`.

**Comment by ixmanuel** [comment]
Yes, removing ResourceAsInput is counter intuitive.

**Comment by ixmanuel** [comment]
@yegor256 InputOf done.

File: src/main/java/org/cactoos/io/InputOf.java [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+    public InputOf(final InputStream stream) {
+        this(() -> stream);
+    }
+
+    /**
+     * Ctor.
+     *
+     * @param input The input
+     */
+    public InputOf(final Input input) {
```
Comment (yegor256) [inline src/main/java/org/cactoos/io/InputOf.java]: @ixmanuel let's make it `private`

**Comment by yegor256** [comment]
@ixmanuel looks good, one small comment above

**Comment by ixmanuel** [comment]
@yegor256, Now the main constructor is private.

**Comment by yegor256** [comment]
@rultor try to merge

**Comment by ixmanuel** [comment]
@yegor256, new test and fixed an independent ctor.

**Comment by yegor256** [comment]
@ixmanuel but the branch is merged already! Or I don't understand what's going on :)

**Comment by ixmanuel** [comment]
@yegor256 I updated this almost at the same time you published it. Then I added some more improvements. Let me close this and open another one. Thanks.

**Comment by ixmanuel** [comment]
@yegor256 Note: If you are reviewing now, BytesOf #340 and ReplacedAllText #341 are in synch with the branch you merged not with these post-updates. Now I'm creating a new pull request for theses post-updates of InputOf.

**Comment by 0crat** [comment]
Oops! Job `gh:yegor256/cactoos#312` was not in scope

**Pull request closed without merging**